src/api/environment.cc

```diff
@@ -52,8 +52,7 @@ bool AllowWasmCodeGenerationCallback(Local<Context> context,
 bool ShouldAbortOnUncaughtException(Isolate* isolate) {
   DebugSealHandleScope scope(isolate);
   Environment* env = Environment::GetCurrent(isolate);
-  return env != nullptr &&
-         (env->is_main_thread() || !env->is_stopping()) &&
+  return env != nullptr && (env->is_main_thread() || env->can_call_into_js()) &&
```

Comment:
Why this change? This is here to ensure that the uncaught termination exception generated as part of stopping the environment does not result in the process aborting entirely, so `is_stopping()` is the right flag to check here.

Reply:
Yeah, the semantics of two flags should still be distinguishable by their names. 

I've updated the PR to fix that the broken implication "when the env is stopping, it should not be able to call into js" and ensured thread-safety in `Environment::ExitEnv`.